src/vs/workbench/electron-browser/workbench.ts

```diff
@@ -1160,7 +1177,19 @@ export class Workbench extends Disposable implements IPartService {
 	isVisible(part: Parts): boolean {
 		switch (part) {
 			case Parts.TITLEBAR_PART:
-				return this.getCustomTitleBarStyle() === 'custom' && (!browser.isFullscreen() || this.menubarVisibility === 'visible' || this.menubarVisibility === 'toggle');
+				if (this.getCustomTitleBarStyle() !== 'custom') {
```

Comment:
@sbatten we never want the custom title bar on mac, or do we? So I would also hardcode to `return false` if `isMacintosh` and then check for the setting.

Reply:
Sorry ignore that comment, we want the custom title bar on macOS but not the menu, but both are in one part now so this makes sense.

Reply:
Custom titlebar is the default on macOS. The menu is what we never want.